kvs/src/main.rs

```diff
@@ -1,35 +1,36 @@
 #![feature(never_type)]
```

Comment:
Not gonna try to comment constructively here since I don't know this code well but I'm surprised to see so many changes! Should this be in a separate PR?